app/src/main/java/com/kylehanish/todo/MainActivity.java

```diff
@@ -0,0 +1,237 @@
+import com.kylehanish.todo.fragments.TodoDialogFragment;
+import com.kylehanish.todo.interfaces.iTodoItemChangeListener;
+import com.kylehanish.todo.repository.TodoSQLRepository;
+import com.kylehanish.todo.repository.iTodoRepository;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import butterknife.Unbinder;
+
+public class MainActivity extends AppCompatActivity implements iTodoItemChangeListener {
+
+
+    private static final String BUNDLE_TODO_ITEMS = "todo_items";
+
+//    Views
+
+    @BindView(R.id.items_list)
+    RecyclerView recyclerView;
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.empty_todo_list)
+    TextView emptyTodoList;
+
+    //    Page Variables
+    private Context mContext;
+    private Unbinder mUnbinder;
+    private TodoItemArrayAdapter mAdapter;
+    private ArrayList<TodoItem> mTodoItems;
+    private iTodoRepository mRepository;
+    private LinearLayoutManager layoutManager;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        mUnbinder = ButterKnife.bind(this);
+
+        mContext = this;
+
+        setSupportActionBar(toolbar);
+
+        mRepository = new TodoSQLRepository(mContext);
+
+        SetBundleValues(savedInstanceState);
+        getListData();
+
+        mAdapter = new TodoItemArrayAdapter(this, R.layout.list_item_todo, mTodoItems, this);
+        recyclerView.setAdapter(mAdapter);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+//        recyclerView.setOnLongClickListener(new View.OnLongClickListener() {
+//            @Override
+//            public boolean onLongClick(View v) {
+//                if (getFragmentManager().findFragmentByTag(PreviewDialogFragment.TAG) == null) {
+//                    PreviewDialogFragment dialogFragment = new PreviewDialogFragment();
+//
+//                    int position = recyclerView.getChildLayoutPosition(v);
+//                    TodoItem item = mTodoItems.get(position);
+//
+//                    if (item != null) {
+//                        Bundle args = new Bundle();
+//                        args.putParcelable(PreviewDialogFragment.BUNDLE_TODO_ITEM, item);
+//                        dialogFragment.setArguments(args);
+//                    }
+//
+//                    dialogFragment.show(getFragmentManager(), PreviewDialogFragment.TAG);
+//                }
+//
+//                return true;
+//            }
+//        });
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
```

Comment:
You can `switch item.getItemId()` and `case R.id.add_new_item`

Reply:
Nice catch. That would make cleaner.